starknet_py/net/models/transaction.py

```diff
@@ -1,28 +1,245 @@
+    ) -> Dict[str, Any]:
+        return decompress_program(data=data, many=many)
+
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network.
+        """
+        return compute_declare_transaction_hash(
+            contract_class=self.contract_class,
+            chain_id=chain_id.value,
+            sender_address=self.sender_address,
+            max_fee=self.max_fee,
+            version=self.version,
+            nonce=self.nonce,
+        )
+
+
+@dataclass(frozen=True)
+class DeployAccount(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is a deployment of a StarkNet account
+    contract.
+    """
+
+    class_hash: int = field(metadata={"marshmallow_field": Felt()})
+    contract_address_salt: int = field(metadata={"marshmallow_field": Felt()})
+    constructor_calldata: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    version: int = field(metadata={"marshmallow_field": Felt()})
+
+    @classmethod
+    def tx_type(cls) -> TransactionType:
+        return TransactionType.DEPLOY_ACCOUNT
+
+    @marshmallow.post_dump
+    def post_dump(self, data: Dict[str, Any], many: bool, **kwargs) -> Dict[str, Any]:
+        data["type"] = "DEPLOY_ACCOUNT"
+        return data
+
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network.
+        """
+        contract_address = compute_address(
+            salt=self.contract_address_salt,
+            class_hash=self.class_hash,
+            constructor_calldata=self.constructor_calldata,
+            deployer_address=0,
+        )
+        return compute_deploy_account_transaction_hash(
+            version=self.version,
+            contract_address=contract_address,
+            class_hash=self.class_hash,
+            constructor_calldata=self.constructor_calldata,
+            max_fee=self.max_fee,
+            nonce=self.nonce,
+            salt=self.contract_address_salt,
+            chain_id=chain_id.value,
+        )
+
+
+@dataclass(frozen=True)
+class InvokeFunction(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is an invocation of a Cairo contract
+    function.
+    """
+
+    contract_address: int = field(metadata={"marshmallow_field": Felt()})
+    calldata: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    nonce: Optional[int] = field(metadata={"marshmallow_field": NoneFelt()})
+    # A field element that encodes the signature of the invoked function.
+    # The entry_point_selector is deprecated for version 1 and above (transactions
+    # should go through the '__execute__' entry point).
+    entry_point_selector: Optional[int] = field(
+        default=None, metadata={"marshmallow_field": NoneFelt()}
+    )
```

Comment:
It it's only for version 0, we shouldn't have it here probably

Reply:
Though we still support tx version 0 so we can't exactly remove that.